draco/analysis/dayenu.py

```diff
@@ -189,6 +188,147 @@ def _get_cut(self, prod):
+        Otherwise will construct a filter for all unique single-time
+        frequency masks (can be significantly slower).  Default is True.
+    atten_threshold : float
+        Mask any frequency where the diagonal element of the filter
+        is less than this fraction of the median value over all
+        unmasked frequencies.  Default is 0.0 (i.e., do not mask
+        frequencies with low attenuation).
+    """
+
+    epsilon = config.Property(proptype=float, default=1e-12)
+    tauw = config.Property(proptype=float, default=0.400)
+    single_mask = config.Property(proptype=bool, default=True)
+    atten_threshold = config.Property(proptype=float, default=0.0)
+
+    def process(self, stream):
+        """Filter delays below some cutoff.
+
+        Modifies container in place.
+
+        Parameters
+        ----------
+        stream : TimeStream or SiderealStream
+            Raw visibilities.
+
+        Returns
+        -------
+        stream_filt : TimeStream or SiderealStream
+            Filtered visibilities.
+        """
+        # Distribute over products
+        stream.redistribute(["ra", "time"])
+
+        # Extract the required axes
+        freq = stream.freq[:]
+
+        ntime = stream.vis.local_shape[2]
+
+        # Dereference the required datasets
+        vis = stream.vis[:].local_array
+        weight = stream.weight[:].local_array
+
+        # Identify baselines that are always flagged
+        temp = np.any(weight > 0.0, axis=(0, 2))
+        baseline_flag = np.zeros_like(temp)
+        self.comm.Allreduce(temp, baseline_flag, op=MPI.LOR)
+        baseline_mask = ~baseline_flag
+
+        if np.all(baseline_mask):
+            self.log.error("All data flaged as bad.")
+            return None
+
+        if np.any(baseline_mask) and not self.single_mask:
+            valid = np.flatnonzero(baseline_flag)
+        else:
+            valid = slice(None)
+
+        # Loop over products
+        for tt in range(ntime):
+
+            t0 = time.time()
+
+            # Flag frequencies and times with zero weight
+            if self.single_mask:
+                flag = (weight[:, :, tt] > 0.0) | baseline_mask
+                flag = np.all(flag, axis=-1, keepdims=True)
+                weight[:, :, tt] *= flag.astype(weight.dtype)
+            else:
+                flag = weight[:, valid, tt] > 0.0
+
+            if not np.any(flag):
+                continue
+
+            tvis = np.ascontiguousarray(vis[:, valid, tt])
+            tvar = tools.invert_no_zero(weight[:, valid, tt])
+
+            self.log.debug(f"Filter time {tt} of {ntime}. [{self.tauw:0.3f} microsec]")
+
+            # Construct the filter
+            try:
+                NF, index = highpass_delay_filter(
+                    freq, self.tauw, flag, epsilon=self.epsilon
+                )
+
+            except np.linalg.LinAlgError as exc:
+                self.log.error(f"Failed to converge while processing time {tt}: {exc}")
```

Comment:
I'm not too familiar with the behaviour of the dayenu filter - is it expected to throw `LinAlgError` every now and then? Or would it be better for the pipeline to fail here if there's an issue?

Reply:
Sounds good